packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -181,130 +194,279 @@ class CupertinoAlertDialog extends StatelessWidget {
+
+    // Set our size now that layout calculations are complete.
+    size = new Size(_kCupertinoDialogWidth, dialogHeight);
+
+    // Set the position of the actions box to sit at the bottom of the dialog.
+    // The content box defaults to the top left, which is where we want it.
+    assert(actions.parentData is MultiChildLayoutParentData);
+    final MultiChildLayoutParentData actionParentData = actions.parentData;
+    actionParentData.offset = new Offset(0.0, contentSize.height);
+  }
+
+  _DialogChildren _findDialogChildren() {
+    RenderBox content;
+    RenderBox actions;
+    final List<RenderBox> children = getChildrenAsList();
+    for (RenderBox child in children) {
+      final MultiChildLayoutParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSections.contentSection) {
+        content = child;
+      } else if (parentData.id == _AlertDialogSections.actionsSection) {
+        actions = child;
+      }
     }
+    assert(content != null);
+    assert(actions != null);
 
-    final double textScaleFactor = MediaQuery.textScaleFactorOf(context);
-    return new GestureDetector(
-      onTap: onPressed,
-      behavior: HitTestBehavior.opaque,
-      child: new Container(
-        alignment: Alignment.center,
-        padding: new EdgeInsets.all(8.0 * textScaleFactor),
-        child: new DefaultTextStyle(
-          style: style,
-          child: child,
-          textAlign: TextAlign.center,
-        ),
-      ),
+    return new _DialogChildren(
+      content: content,
+      actions: actions,
     );
   }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    defaultPaint(context, offset);
+  }
+
+  @override
+  bool hitTestChildren(HitTestResult result, { Offset position }) {
+    return defaultHitTestChildren(result, position: position);
+  }
+}
+
+// Visual components of an alert dialog that need to be explicitly sized and
+// laid out at runtime.
+enum _AlertDialogSections {
+  contentSection,
+  actionsSection,
 }
 
-// Constructs a text content section typically used in a CupertinoAlertDialog.
+// Data structure used to pass around references to multiple dialog pieces for
+// layout calculations.
+class _DialogChildren {
+  _DialogChildren({
+    this.content,
+    this.actions,
+  });
+
+  final RenderBox content;
+  final RenderBox actions;
+}
+
+// The "content section" of a CupertinoAlertDialog.
 //
 // If title is missing, then only content is added.  If content is
 // missing, then only title is added. If both are missing, then it returns
 // a SingleChildScrollView with a zero-sized Container.
-class _CupertinoAlertTitleSection extends StatelessWidget {
-  const _CupertinoAlertTitleSection({
+class _CupertinoAlertContentSection extends StatelessWidget {
+  const _CupertinoAlertContentSection({
     Key key,
     this.title,
     this.content,
     this.scrollController,
-  }) : super(key: key);
+  }) : super(key: const Key('this_is_the_root'));
```

Comment:
There's still one here

Reply:
Oops. Good catch. I had added that temporarily to verify something and then forgot to remove it again.  I've updated the PR with that key removed.